src/main/kotlin/com/recipebook/domain/ingredient/Ingredient.kt

```diff
@@ -0,0 +1,23 @@
+package com.recipebook.domain.ingredient
+
+import com.recipebook.domain.recipe.Recipe
+import com.recipebook.orm.AbstractJpaPersistable
+import java.io.Serializable
+import javax.persistence.Entity
+import javax.persistence.ManyToMany
+
+@Entity
+class Ingredient(var name: String, var unit: String, var quantity: Double) : AbstractJpaPersistable<Ingredient>(),
+        Serializable {
+     }
+
+    @ManyToMany
+    val ingredients: List<Recipe> = mutableListOf()
+
+    fun addIngredient(ingredient: Ingredient) {
+        if (!ingredients.contains(ingredient)) {
+
```

Comment:
Czemu puste?

Reply:
podpowiedź: nie działa bo ta metoda to powinno być "addRecipe" - ze sprawdzeniem czy przepis jest na liście przepisów

Reply:
Chyba Ingredient jest w przepisie. To jest kontener dla Ingredient a nie Recipe

Reply:
Ingredient może być w wielu Recipe, Recipe może mieć wiele składników.
klasa Recipe posiada więc listę Ingredient, które posiada,
klasa Ingredient posiada listę Recipe, które go zawierają

Tu operujesz na klasie Ingredient, więc tworzysz listę Recipe i potrzebujesz tez metody addRecipe itd